src/pack_unpack.c

```diff
@@ -348,6 +348,33 @@ static json_t *pack_string(scanner_t *s, va_list *ap)
     }
 }
 
+static json_t *pack_object_inter(scanner_t *s, va_list *ap, int need_incref)
+{
+    int nullable;
+    json_t *json;
+
+    next_token(s);
+    nullable = token(s) == '?';
+    if (!nullable)
+        prev_token(s);
+
+    json = va_arg(*ap, json_t *);
+
+    if (json) {
+        return need_incref ? json_incref(json) : json;
+    }
+    if (nullable) {
+        return json_null();
+    } else if (s->next_token.token == '*') {
```

Comment:
The previous branch return's you could drop the `else` and just start a new branch.  IMO this makes the code easier to read.

Reply:
I rewrote to switch, look, that's better?